imap_processing/ultra/l1b/de.py

```diff
@@ -0,0 +1,53 @@
+"""Calculates Annotated Direct Events."""
+
+import numpy as np
+
+
+def calculate_de(data_dict):
+    """
+    Create dictionary with defined datatypes for Direct Event Data.
```

Comment:
Similar point - if you're creating a dictionary with the same keys every time, a dataclass will provide some nice convenience factors, while also being more stable and safe than a dictionary (by providing tracing back to what the keys should be, providing additional documentation for each key, and by disallowing someone to update the dictionary later.) Feel free to message me on slack for a zoom call if you want to discuss.

Reply:
This would be instead of the dataset suggestion by Greg - a dataset here would also function in a very similar way so it would also solve a lot of the same issues as a dataclass will. So, it comes down to you being the expert to figure out which way works the best.

Reply:
@maxinelasp I saw your post after I had changed it. What I think I will do is leave it for now, but if you use a dataclass for MAG I'll take a look and maybe make modifications.

Reply:
A dataset functions very similarly in this case, and in some ways is a better match than a dataclass. I can walk you though my style of using a dataclass, and then you can decide which you like better. 

A dataset is more closely aligned with the inputs and outputs, which is nice. A dataclass you have a bit more control over the structure and on expanding the variables (or, including variables that are not in the output, things like that) so it does come down to a style choice at that point. Both are better than dicts!